presto-main-base/src/main/java/com/facebook/presto/execution/QueryTracker.java

```diff
@@ -220,13 +221,24 @@ private void enforceTimeLimits()
                 continue;
             }
             Duration queryMaxRunTime = getQueryMaxRunTime(query.getSession());
+            Duration queryMaxQueuedTime = getQueryMaxQueuedTime(query.getSession());
             QueryLimit<Duration> queryMaxExecutionTime = getMinimum(
                     createDurationLimit(getQueryMaxExecutionTime(query.getSession()), QUERY),
                     query.getResourceGroupQueryLimits()
                             .flatMap(ResourceGroupQueryLimits::getExecutionTimeLimit)
                             .map(rgLimit -> createDurationLimit(rgLimit, RESOURCE_GROUP)).orElse(null));
             long executionStartTime = query.getExecutionStartTimeInMillis();
             long createTimeInMillis = query.getCreateTimeInMillis();
+            long queuedTimeInMillis;
+            if (executionStartTime > 0) {
+                queuedTimeInMillis = executionStartTime - createTimeInMillis;
+            }
+            else {
+                queuedTimeInMillis = currentTimeMillis() - createTimeInMillis;
```

Comment:
Supposing this were set to a smaller value, wouldn't this also include slowness for other non-queueing reasons, such as a slow metastore when doing table resolution and permission checks etc?

Reply:
Good catch. You are right. With the way, we currently compute `queued_time`, it will include everything from `creation_time` to `execution_time` which also include things like dispatching time, planning time etc. 

Diving deeper into the code, I see that QueryStateTimer already computes `queued_time` accurately [here](https://github.com/prestodb/presto/blob/master/presto-main-base/src/main/java/com/facebook/presto/execution/QueryStateTimer.java#L280-L283) which can be exposed and used while doing this check. Will be updating the PR to use this metric directly instead of computing it here.

Reply:
@tdcmeehan : Gentle ping!!